### PR TITLE
Post/using focus style for accessibility

### PR DESCRIPTION
**Pull request by Emilsone**
Hi @ericwbailey, I have made the required changes as mentioned. Please kindly review and leave your feedback

### CODE REVIEW

**Comment by Emilsone** [comment]
Hello @ericwbailey,   I still haven't gotten any response from you  concerning the new changes I made?

File: src/posts/2021-09-27-Using-focus-style-for-accessibility [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+
+It is wrong to remove the focus indicator as it makes your website inaccessible for people navigating with the keyboard the best way is to apply a different focus style for mouse/pointer users.
+
+code example
+```CSS
+:focus:not(:focus-visible) {
+ outline:none
+}
+```
+This will hide the focus indicator if the element receives focus via the mouse, but it will show up on the keyboard when tabbing.
```
Comment (ericwbailey) [inline src/posts/2021-09-27-Using-focus-style-for-accessibility:97]:
```suggestion

This will hide the focus indicator if the element receives focus via the mouse, but it will show up on the keyboard when tabbing.
```

File: src/posts/2021-09-27-Using-focus-style-for-accessibility [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+
+## Focus-within
+The pseudo-class:focus-within is used to style the parent element of focusable elements it selects an element that has focus or when a descendant has focus, it is very useful when designing forms, supported in modern browsers.
```
Comment (ericwbailey) [inline src/posts/2021-09-27-Using-focus-style-for-accessibility:100]:
```suggestion
## Focus-within

The pseudo-class:focus-within is used to style the parent element of focusable elements it selects an element that has focus or when a descendant has focus, it is very useful when designing forms, supported in modern browsers.
```

File: src/posts/2021-09-27-Using-focus-style-for-accessibility [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+</form>
+```
+```CSS
+.form-container:focus-within {
+ background: #ff8;
+ color: black;
+ padding:20px;
+}
+```
+In the above example, we created a styling for the form element the background color changes when the input receives focus or when the user clicks on the input.
```
Comment (ericwbailey) [inline src/posts/2021-09-27-Using-focus-style-for-accessibility:117]:
```suggestion

In the above example, we created a styling for the form element the background color changes when the input receives focus or when the user clicks on the input.
```

File: src/posts/2021-09-27-Using-focus-style-for-accessibility [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+## Summary
+Having a focus indicator on your website will improve the usability and user experience and also make it accessible to all users.
```
Comment (ericwbailey) [inline src/posts/2021-09-27-Using-focus-style-for-accessibility:119]:
```suggestion

## Summary

Having a focus indicator on your website will improve the usability and user experience and also make it accessible to all users.
```

**Review by ericwbailey — CHANGES_REQUESTED** [review]
@Emilsone I added more feedback around Markdown formatting, to ensure the content will render properly. Additionally, I noticed there was still unaddressed feedback from the last round of review.

Additionally, I would like to see terms like "above" and "below" replaced with "following" and "preceding", as it is technically a violation of [WCAG Success Criterion 1.3.3: Sensory Characteristics](https://www.w3.org/WAI/WCAG21/Understanding/sensory-characteristics.html).

**Comment by Emilsone** [comment]
Hi @ericwbailey, I have made the old and new changes you requested please kindly review them? Thanks for the nice feedback

File: src/posts/2021-09-27-Using-focus-style-for-accessibility (hunk 1 above)
Comment (Emilsone) [reply src/posts/2021-09-27-Using-focus-style-for-accessibility:106]: Thanks @ericwbailey  i have fixed the required changes please kindly review

**Comment by Emilsone** [comment]
> Hi @ericwbailey, I have made the old and new changes you requested please kindly review them? Thanks for the nice feedback

File: src/posts/2021-09-27-Using-focus-style-for-accessibility [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+    year: 2016
+  - title: Never remove CSS outlines
+    url: https://www.a11yproject.com/posts/2013-01-25-never-remove-css-outlines/
+    source: Guilherme Simões
+    year: 2013
+tags:
+  - How-to
+---
+
+While accessing any digital interface, a focus indicator is important it highlights the active intractable component in a webpage.
```
Comment (ericwbailey) [inline src/posts/2021-09-27-Using-focus-style-for-accessibility:24]:
```suggestion
A focus indicator is important because it highlights the active intractable component in a webpage. 
```

Shifting this to active tone. "Accessing a digital interface" is implied here, as well, so I'm suggesting removing it.

File: src/posts/2021-09-27-Using-focus-style-for-accessibility [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+    year: 2013
+tags:
+  - How-to
+---
+
+While accessing any digital interface, a focus indicator is important it highlights the active intractable component in a webpage. 
+
+## What is a focus style?
+
+A focus is a pseudo-class used to indicate or highlights an intractable component in a webpage.
```
Comment (ericwbailey) [inline src/posts/2021-09-27-Using-focus-style-for-accessibility:28]: >A focus

A focus what?

Comment (ericwbailey) [reply src/posts/2021-09-27-Using-focus-style-for-accessibility:28]: Can we link "a pseudo-class" to [this MDN article on `:focus`](https://developer.mozilla.org/en-US/docs/Web/CSS/:focus). That will help explain the jargon to someone not familiar with the concept.

File: src/posts/2021-09-27-Using-focus-style-for-accessibility [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+
+## Why is it important?
+
+A focus style is necessary because some people use a mouse or a touch device to access a website some rely on using a keyboard to navigate on a website. Having a focus indicator will improve the usability of your website and makes it accessible, It also tells us the interactive element we are focused on.
```
Comment (ericwbailey) [inline src/posts/2021-09-27-Using-focus-style-for-accessibility:32]: Can we revise the first sentence of this paragraph? It doesn't make sense to me when read aloud.

Comment (ericwbailey) [reply src/posts/2021-09-27-Using-focus-style-for-accessibility:32]: >Having a focus indicator will improve the usability of your website and makes it accessible,

How does a focus indicator accomplish this? We'll need to explain a statement like that to the reader, especially if they're unfamiliar with what a focus indicator is.

File: src/posts/2021-09-27-Using-focus-style-for-accessibility [hunk 8]
```diff
@@ -0,0 +1,133 @@
...
+
+## How to apply focus style
+
+Preceding code is an example of how we can apply focus style to our elements:
```
Comment (ericwbailey) [inline src/posts/2021-09-27-Using-focus-style-for-accessibility:36]:
```suggestion
The following code is an example of how we can apply focus style to our elements:
```

Preceding is what comes before, following is what comes after.

File: src/posts/2021-09-27-Using-focus-style-for-accessibility [hunk 9]
```diff
@@ -0,0 +1,133 @@
...
+
+### Applying focus style to a link
+
+```CSS
+a:focus {
+  outline: 5px solid red;
+  }
```
Comment (ericwbailey) [inline src/posts/2021-09-27-Using-focus-style-for-accessibility:43]:
```suggestion
a:focus {
  outline: 5px solid red;
}
```

File: src/posts/2021-09-27-Using-focus-style-for-accessibility [hunk 10]
```diff
@@ -0,0 +1,133 @@
...
+
+## How to apply focus style
+
+Preceding code is an example of how we can apply focus style to our elements:
+
+### Applying focus style to a link
+
+```CSS
+a:focus {
+  outline: 5px solid red;
```
Comment (ericwbailey) [inline src/posts/2021-09-27-Using-focus-style-for-accessibility:42]: Can we use `rem` instead of `px`? `px` won't scale for some screen magnification scenarios.

File: src/posts/2021-09-27-Using-focus-style-for-accessibility [hunk 11]
```diff
@@ -0,0 +1,133 @@
...
+  }
+```
+
+The following code example is a link that has a custom style for it focus state:
```
Comment (ericwbailey) [inline src/posts/2021-09-27-Using-focus-style-for-accessibility:46]:
```suggestion
The following code example is a link that has a custom style for its focus state:
```

Comment (ericwbailey) [reply src/posts/2021-09-27-Using-focus-style-for-accessibility:46]: Are you referring to the previous code example starting on line 40?

File: src/posts/2021-09-27-Using-focus-style-for-accessibility [hunk 12]
```diff
@@ -0,0 +1,133 @@
...
+
+### Code explanation
+
+In the following code example, we added focus to our link and gave it an outline of solid red with a value of 5px. Please note the focused style is applied only to the link tag itself.
```
Comment (ericwbailey) [inline src/posts/2021-09-27-Using-focus-style-for-accessibility:50]: I am confused about this and the previous paragraph. I think they need to be combined.

File: src/posts/2021-09-27-Using-focus-style-for-accessibility [hunk 13]
```diff
@@ -0,0 +1,133 @@
...
+
+The preceding code is another example for applying focus to a form input:
```
Comment (ericwbailey) [inline src/posts/2021-09-27-Using-focus-style-for-accessibility:52]:
```suggestion
The following code is another example for applying focus to a form input:
```

File: src/posts/2021-09-27-Using-focus-style-for-accessibility [hunk 14]
```diff
@@ -0,0 +1,133 @@
...
+
+### Applying focus style to a form input
+
+```HTML
+<form>
+<label for="email"> Email :</label>
+<input class="email-input" value="Click me to see my focus state">
+</form>
```
Comment (ericwbailey) [inline src/posts/2021-09-27-Using-focus-style-for-accessibility:60]:
```suggestion
<form>
  <label for="email">Email:</label>
  <input class="email-input" value="Click me to see my focus state">
</form>
```

Indenting nested elements, removing unnecessary spaces in the label.

File: src/posts/2021-09-27-Using-focus-style-for-accessibility [hunk 15]
```diff
@@ -0,0 +1,133 @@
...
+```
+
+```CSS
+.email-input:focus {
+ background-color: gray;
+ color: black;
+}
+```
+
+You can create your custom outline style aside from the blue outline default style in focus. Always ensure the color contrast for your focus indicator is sufficient especially important for people with low vision. This will also benefit those using their screen outside in a very bright space.
```
Comment (ericwbailey) [inline src/posts/2021-09-27-Using-focus-style-for-accessibility:70]: The default isn't always blue. It depends on the device, its operating system, and the browser. For example, my system theme highlight color is set to gray, so I see a gray focus ring for browsers that support it. 

Can we rephrase this to something like, "the browser's default focus style." It also might be worth mentioning Firefox, which uses a focus ring that is too subtle to be used reliably.

Comment (ericwbailey) [reply src/posts/2021-09-27-Using-focus-style-for-accessibility:70]: We'll also probably want to link to a color contrast checker like [WebAIM's link contrast checker](https://webaim.org/resources/linkcontrastchecker/), to help people new to the concept of color contrast know what it is and how to check their ratios.

File: src/posts/2021-09-27-Using-focus-style-for-accessibility [hunk 16]
```diff
@@ -0,0 +1,133 @@
...
+
+It is wrong to remove the focus indicator as it makes your website inaccessible for people navigating with the keyboard the best way is to apply a different focus style for mouse/pointer users.
+
+Code example:
+
+```CSS
+:focus:not(:focus-visible) {
+ outline:none
+}
+```
```
Comment (ericwbailey) [inline src/posts/2021-09-27-Using-focus-style-for-accessibility:104]: I would like to revise this code example to use something other than `:focus:not(:focus-visible)`. I consider this approach an antipattern, as it adjusts focus behavior to something unexpected. This has created confusion for people with lower tech literacy who rely on focus indicators, as well as creates a separate class of experience for non-mouse users, which isn't good.

**Review by ericwbailey — CHANGES_REQUESTED** [review]

**Comment by ericwbailey** [comment]
Closing due to lack of activity.
